### PR TITLE
Update NNCE LTT only on state change

### PR DESCRIPTION
**Pull request by rhrazdil**
**Is this a BUG FIX or a FEATURE ?**:
/kind bug

**What this PR does / why we need it**:
Update NNCe Last Transition Time only when state
of a condition is changed.

**Special notes for your reviewer**:

**Release note**:

```release-note
update Last Transition Time only when condition state actually changes
```

### CODE REVIEW

File: vendor/github.com/nmstate/kubernetes-nmstate/api/shared/condition_types.go [hunk 1]
```diff
@@ -50,10 +50,12 @@ func (conditions *ConditionList) Set(conditionType ConditionType, status corev1.
 
 	// If there is different status, reason or message update it
 	if condition.Status != status || condition.Reason != reason || condition.Message != message {
+		if condition.Status != status {
```
Comment (phoracek) [inline vendor/github.com/nmstate/kubernetes-nmstate/api/shared/condition_types.go:53]: We should not apply patches to vendor/, no?

Comment (rhrazdil) [reply vendor/github.com/nmstate/kubernetes-nmstate/api/shared/condition_types.go:53]: We are vendoring the API as a subpackage, so we need to sync changes done in API

Comment (phoracek) [reply vendor/github.com/nmstate/kubernetes-nmstate/api/shared/condition_types.go:53]: I see, thanks

**Review by qinqon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**